pkg/psmdb/backup/backup.go

```diff
@@ -60,13 +74,29 @@ func HasActiveJobs(cl client.Client, cluster, namespace string, current Job) (bo
 		if r.Name == current.Name && current.Type == TypeRestore {
 			continue
 		}
-		if r.Spec.ClusterName == cluster &&
+		if r.Spec.ClusterName == cluster.Name &&
 			r.Status.State != api.RestoreStateReady &&
 			r.Status.State != api.RestoreStateError &&
 			r.Status.State != api.RestoreStateWaiting {
 			return true, nil
 		}
 	}
 
+	pbm, err := NewPBM(cl, cluster)
+	if err != nil {
+		return false, errors.Wrap(err, "getting pbm object")
+	}
+
+	allowLock = append([]LockHeaderPredicate{NotJobLock(current)}, allowLock...)
+
+	hasLocks, err := pbm.HasLocks(allowLock...)
+	if err != nil {
+		return false, errors.Wrap(err, "getting pbm locks")
+	}
+
+	if hasLocks {
+		return true, nil
+	}
+
 	return false, nil
```

Comment:
```suggestion
	return pbm.HasLocks(allowLock...)
```

Reply:
Are you sure? It's shorter, but it's also harder to add something to the end of the function, isn't it?